File: problems/1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
const maxOperations = (nums, k) => {
    nums.sort((a, b) => a - b);
    let l = 0;
    let r = nums.length - 1;
    let res = 0;
    while (l < r) {
        const sum = nums[l] + nums[r];
        if (k === sum) {
            res ++;
            l ++;
            r --;
        } else if (k < sum) {
            r --;
        } else {
            l ++;
        }
    }
    return res;
};

File: problems/1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        l = 0
        r = len(nums) - 1
        res = 0
        nums.sort()
    
        while l < r:
            sum = nums[l] + nums[r]
            if sum < k:
                l += 1
            elif sum > k:
                r -= 1
            else:
                res += 1
                l += 1
                r -= 1

        return res